[payment] prorate team subscriptions

JanKoehnlein opened the pull request:
# Description

Team subscriptions all have unlimited hours, so we can charge/credit prorated prices when the number of seats changes within a subscription period. This will allow us to get away from the seat-based UI and also adds more transparency towards our users.

# How to test

You cannot test this in preview env currently, as payment doesn't work there. I manually ran the API calls against our Chargebee test site

```release-notes
When seats are added / removed to a team subscriptions in the middle of a subscription period, only the prorated price is charged / credited for that period
```

roboquat commented:
@JanKoehnlein: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

roboquat commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **jankoehnlein** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["jankoehnlein"]} -->

JanKoehnlein commented:
/werft run with-payment

JanKoehnlein commented:
From testing on our CB test site: Removed 1 and added two seats in the middle of a subscription period
![Screenshot 2022-01-18 at 17 53 50](https://user-images.githubusercontent.com/377752/149983270-48e300a7-aa0d-4666-bc72-db77ff9803b6.png)

geropl commented on components/server/ee/src/workspace/gitpod-server-impl.ts:
```diff
@@ -1304,7 +1304,7 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         }
         await this.doUpdateSubscription(userId, chargebeeSubscriptionId, {
             plan_quantity: newQuantity,
-            end_of_term: applyEndOfTerm
```
Does this default to `false`? :thinking:

geropl commented:
/hold
/lgtm

Holding in case we do not want to deploy that tomorrow morning @JanKoehnlein

roboquat commented:
LGTM label has been added.  <details>Git tree hash: f053858a9f751dc8fc49f9544b9de871f6e9a587</details>

roboquat commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7663#issuecomment-1015623589" title="LGTM">geropl</a>*

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

JanKoehnlein commented:
/approve no-issue

roboquat commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7663#issuecomment-1015623589" title="LGTM">geropl</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7663#issuecomment-1020225868" title="Approved">JanKoehnlein</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/7663#issuecomment-1020225868" title="Approved">JanKoehnlein</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [JanKoehnlein,geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

JanKoehnlein commented:
I am closing this PR as we decided to not change anything for existing customers but implement the new team plans next to the old one. We should nevertheless use proration for the new implementation
